src/power_grid_model_io/converters/tabular_converter.py

```diff
@@ -178,24 +178,29 @@ def _convert_table_to_component(
         if table not in data:
             return None
 
-        n_records = len(data[table])
+        if "filter" in attributes:
+            table_mask = self._parse_table_filters(data=data, table=table, filtering_functions=attributes["filter"])
+        else:
+            table_mask = np.ones(len(data[table]), dtype=bool)
 
         try:
-            pgm_data = initialize_array(data_type=data_type, component_type=component, shape=n_records)
+            pgm_data = initialize_array(data_type=data_type, component_type=component, shape=np.sum(table_mask))
         except KeyError as ex:
             raise KeyError(f"Invalid component type '{component}' or data type '{data_type}'") from ex
 
         if "id" not in attributes:
             raise KeyError(f"No mapping for the attribute 'id' for '{component}s'!")
 
         # Make sure that the "id" column is always parsed first (at least before "extra" is parsed)
-        sorted_attributes = sorted(attributes.items(), key=lambda x: "" if x[0] == "id" else x[0])
+        attributes_without_filter = {k: v for k, v in attributes.items() if k != "filter"}
+        sorted_attributes = sorted(attributes_without_filter.items(), key=lambda x: "" if x[0] == "id" else x[0])
```

Comment:
filter is usually about _including_ not _excluding_ (except it explicitly is mentioned as `exclude_filter` or just `exclude` or something similar). shouldn't this be the other way around to make it `filtered_attributes`?

Reply:
I like the idea about renaming this functionality to `exclude`. Ill do so.
This would then be named: `attributes_without_exclude_key`.

Reply:
> filter is usually about including not excluding (except it explicitly is mentioned as exclude_filter or just exclude or something similar).

Hmm, not sure if I'd 100% agree there XD How would you understand terms like "filter out" then? I don't think there is a common sense of what filter would be including or excluding, it could even mean transforming. 

In our case, we are more thinking the purpose of sifting stuff like strainers, which is a perfect filtering process. Therefore, I would prefer Nitish's original wording of `attributes_without_filter `

Reply:
python has a built-in `filter` function that explicitly includes objects https://docs.python.org/3/library/functions.html#filter

the ambiguity is bad enough as is, but actually having name conflicts with python built-ins is even worse IMO